crates/wast/src/lib.rs

```diff
@@ -172,6 +177,14 @@ impl Error {
             _ => None,
         }
     }
+
+    /// Returns the underlying custom `&str`, if any, that describes this error.
+    pub fn custom_error(&self) -> Option<&str> {
```

Comment:
Could this be renamed to perhaps `message`? I think that this should be implementable for lex errors as well. The term "custom" is mostly specific to the internals of this file and doesn't necessarily make a ton of sense to consumers.